client/src/modules/dashboard/Events/pages/EventPage.tsx

```diff
@@ -39,18 +39,20 @@ export const EventPage: NextPage = () => {
 
   const confirmRSVP =
     ({ eventId, userId }: MutationConfirmRsvpArgs) =>
-    () => {
-      return confirm().then(() =>
-        confirmRsvpFn({ variables: { eventId, userId } }),
-      );
+    async () => {
+      const ok = await confirm();
+      if (ok) {
+        return confirmRsvpFn({ variables: { eventId, userId } });
+      }
```

Comment:
```suggestion
      if (ok)  confirmRsvpFn({ variables: { eventId, userId } });
```
Since we don't use the return value, we can compress this slightly.